library/std/src/primitive_docs.rs

```diff
@@ -807,10 +807,10 @@ mod prim_tuple {}
 ///
 /// Additionally, `f32` can represent some special values:
 ///
-/// - -0.0: IEEE 754 floating point numbers have a bit that indicates their sign, so -0.0 is a
-///   possible value. For comparison `-0.0 == +0.0` is true but floating point operations can
-///   carry the sign bit through arithmetic operations. This means `-1.0 * 0.0` produces -0.0 and
-///   a negative number rounded to a value smaller than a float can represent also produces -0.0.
+/// - −0.0: IEEE 754 floating point numbers have a bit that indicates their sign, so −0.0 is a
+///   possible value. For comparison `-0.0 == 0.0` is true but floating point operations can
+///   carry the sign bit through arithmetic operations. This means `-1.0 * 0.0` produces −0.0 and
```

Comment:
So, individually I understand and would like either change, but making both at once makes it so that all of the instances of {-,−}0.0 have a subtle difference between them that is below easy notice or, if it gets noticed, is distracting and could lead someone to a possibly incorrect conclusion. My reasoning for writing this in this manner was that the expression is not in a \`\`\`rust code context and therefore its Rust evaluation validity is not necessarily relevant, but that providing the plus sign would add logical clarity, and a learning programmer could trivially infer the intended meaning.

As an alternative: perhaps remove the backticks here and treat it as purely symbolic expressions for consistency, pointedly avoiding Rust's arithmetic operators, e.g.
```suggestion
///   possible value. For comparison −0.0 ≡ +0.0, but floating point operations can carry
///   the sign bit through arithmetic operations. This means −0.0 × +0.0 produces −0.0 and
```

Reply:
I prefer the valid expression `-0.0 == 0.0` that was the old version of the doc, as someone can just try it out, and the meaning is very clear even without the plus sign which makes it invalid in Rust. In effect, that means that in Rust comparing `-0.0` to `0.0` yields equality. I find the ≡ symbol more confusing, as I understand it to mean "is identical to", but -0 is not *identical* to +0; and I find that symbol can have different meanings to different people.

Reply:
Yeah, it's true. IEEE754 does have strange and seemingly self-contradictory opinions about whether or not these two numbers are actually identical or not. But a simple equals would be fine also. Alas, Rust has overloaded that meaning with assignment. IEEE754 renders it almost exactly so, though, for the same reason of disambiguation: "Comparisons shall ignore the sign of zero (so +0 = −0)."

But mmm. I am not sure why use the variant minus sign elsewhere, then? If the desire is for the values to be copyable from the text, then having the "alternate" −0.0 impedes that.

A way of diverting people away from copying the U+2212 symbols would be to move the code we expect to actually run into a \`\`\`rust context. That way it's easily clickable and runnable and also copyable... I have trouble copying small segments from the middle of a sentence and I know many other people who do, also, because they have a shaky hand or other hand-eye coordination problems, making a mouse interaction unwieldy.

Reply:
Oh, I find −0.0 = +0.0 to be good as well: since it's not code it's not assignment anyway. I see two contexts here: text where U+2212 should be used and = is fine for an equality check, and code where U+002D has to be used for minus and == for equality.

That is, I'm now preferring your suggestion but with = instead of ≡.

Reply:
Ahhh~! Then we have consensus.
```suggestion
///   possible value. For comparison −0.0 = +0.0, but floating point operations can carry
///   the sign bit through arithmetic operations. This means −0.0 × +0.0 produces −0.0 and
```